src/git-command-manager.ts

```diff
@@ -229,7 +229,11 @@ class GitCommandManager {
   }
 
   async init(): Promise<void> {
-    await this.execGit(['init', this.workingDirectory])
+    await this.execGit([
+      'init',
+      '--initial-branch=silence_warning_about_default_branch',
```

Comment:
Let's see.

First of all, I have a slight preference to `-b`, and I have a strong preference for `main` instead of that very long branch name.

However, even more important than a short & sweet name is an explanation why this won't break anything. That's really important, and I do not see anything in that regard.

Which means that the most tedious part of this PR is left to the reviewer, which is not really a good time balance.

So let's see where that analysis leads us. According to the the web interface, there is a single caller of that `init()` function, and it is [here](https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/git-source-provider.ts#L85), in context: https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/git-source-provider.ts#L80-L88

That context is _still_ not large enough to see who the branch name might be used (if at all).

A little further down, the default branch is determined: https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/git-source-provider.ts#L106-L117

Even if we moved this before the `init()` call, we could not use it as branch name because `settings.commit` might be set, in which case we're not even on a named branch (and you cannot initialize a Git repository without an initial branch name), and also `settings.ref` might refer to a tag, in which case we _also_ do not have a branch name to work with.

But let's read on.

Here is where the check out _actually_ happens: https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/git-source-provider.ts#L166-L169

That `checkoutInfo.ref` comes from https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/git-source-provider.ts#L148-L155

So let's follow the code to see what that function provides. Here is the code that sets that `ref` attribute: https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L29-L60

In particular, the `ref` attribute is set at the following locations:
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L31
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L36
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L42
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L46
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L51
- https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/src/ref-helper.ts#L54

Most importantly, the `ref` attribute is always set (unless the function throws an exception, in which case the `checkout()` call is not even reached).

Therefore, the initial branch name does not matter (as your patch suggests).

Reply:
I'm happy to use a shorter name. I didn't want to bias a name and confuse people into thinking it mattered, as it doesn't, which is why I picked the really long one. I'm not remotely attached to it.

Would you want more detail in a comment or in a commit message? The file is fairly terse w.r.t. comments, although it would be quite reasonable to include one here (at the risk of it becoming stale if code moves).

I used long form for flags because that seemed to be house style (and to an extent the branch name because everything seems to use long descriptive names):
https://github.com/actions/checkout/blob/b342ec4a9ce8e496606d799796b72dc74766ada5/src/git-command-manager.ts#L161-L163
https://github.com/actions/checkout/blob/b342ec4a9ce8e496606d799796b72dc74766ada5/src/git-command-manager.ts#L174
https://github.com/actions/checkout/blob/b342ec4a9ce8e496606d799796b72dc74766ada5/src/git-command-manager.ts#L177-L185
...

Reply:
> Would you want more detail in a comment or in a commit message?

I was thinking about the commit message, but potentially the "house style" (I like that expression) would prefer it in the PR description. Please note that I am _not_ on the team that develops this GitHub Action, therefore I have no power to merge or reject it. Read: my opinion does not matter _all_ that much here ;-)

> I used long form for flags because that seemed to be house style

Fair enough. When in Rome...